components/server/src/auth/authenticator.ts

```diff
@@ -151,7 +151,7 @@ export class Authenticator {
 
     async deauthorize(req: express.Request, res: express.Response, next: express.NextFunction) {
         const user = req.user;
-        if (!req.isAuthenticated() || !User.is(user)) {
+        if (!req.isAuthenticated() || !User.is(user) || user.blocked) {
             log.info({ sessionId: req.sessionID }, `User is not authenticated.`);
             res.redirect(this.getSorryUrl(`Not authenticated. Please login.`));
```

Comment:
This is not the correct failure mode for blocked users, right?

Reply:
> This is not the correct failure mode for blocked users, right?

Nope, but I think we have bigger problems atm then caring about UX for blocked users.

Let's take and note/follow up PR and merge this one asap. :pray:

Reply:
I wonder if a "cleaner" location for this fix might be:

https://github.com/gitpod-io/gitpod/blob/ed30d9696fd1c0ca5775a2d44d6b4a9e5709ba63/components/server/src/user/user-controller.ts#L147-L150